quickwit-indexing/src/actors/indexing_pipeline.rs

```diff
@@ -543,13 +574,94 @@ mod tests {
     use quickwit_config::IndexingSettings;
     use quickwit_index_config::default_config_for_tests;
     use quickwit_metastore::checkpoint::Checkpoint;
-    use quickwit_metastore::MockMetastore;
+    use quickwit_metastore::{MetastoreError, MockMetastore};
     use quickwit_storage::RamStorage;
     use serde_json::json;
 
     use super::{IndexingPipeline, *};
     use crate::models::IndexingDirectory;
 
+    async fn test_indexing_pipeline_num_fails_before_success(
+        mut num_fails: usize,
+    ) -> anyhow::Result<bool> {
+        quickwit_common::setup_logging_for_tests();
+        let mut metastore = MockMetastore::default();
+        metastore
+            .expect_list_splits()
+            .returning(|_, _, _, _| Ok(Vec::new()));
+        metastore
+            .expect_mark_splits_for_deletion()
+            .returning(|_, _| Ok(()));
+
+        metastore
+            .expect_index_metadata()
+            .withf(|index_id| index_id == "test-index")
+            .returning(move |_| {
+                if num_fails == 0 {
```

Comment:
I wonder if it is possible to do that in mockall with "times".

Basically register the n failure first, and then register the success.